packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -201,86 +182,66 @@ export class RehydratedLocalChannelContext extends LocalChannelContextBase {
 		private readonly snapshotTree: ISnapshotTree,
 	) {
-		super(id, registry, runtime, () => this.services);
-		const blobMap: Map<string, ArrayBufferLike> = new Map<string, ArrayBufferLike>();
-		const clonedSnapshotTree = cloneDeep(this.snapshotTree);
-		// 0.47 back-compat Need to sanitize if snapshotTree.blobs still contains blob contents too.
-		// This is for older snapshot which is generated by loader <=0.47 version which still contains
-		// the contents within blobs. After a couple of revisions we can remove it.
-		if (this.isSnapshotInOldFormatAndCollectBlobs(clonedSnapshotTree, blobMap)) {
-			this.sanitizeSnapshot(clonedSnapshotTree);
-		}
-
-		this.services = new Lazy(() => {
-			return createServiceEndpoints(
-				this.id,
-				dataStoreContext.connected,
-				submitFn,
-				this.dirtyFn,
-				addedGCOutboundReferenceFn,
-				storageService,
-				logger,
-				clonedSnapshotTree,
-				blobMap,
-			);
-		});
-		this.dirtyFn = () => {
-			dirtyFn(id);
-		};
-	}
-
-	public async getChannel(): Promise<IChannel> {
-		if (this.channel === undefined) {
-			this.channel = await this.loadChannel().catch((err) => {
-				throw DataProcessingError.wrapIfUnrecognized(
-					err,
-					"rehydratedLocalChannelContextFailedToLoadChannel",
-					undefined,
+		super(
+			id,
+			runtime,
+			new Lazy(() => {
+				const blobMap: Map<string, ArrayBufferLike> = new Map<string, ArrayBufferLike>();
+				const clonedSnapshotTree = cloneDeep(this.snapshotTree);
+				// 0.47 back-compat Need to sanitize if snapshotTree.blobs still contains blob contents too.
+				// This is for older snapshot which is generated by loader <=0.47 version which still contains
+				// the contents within blobs. After a couple of revisions we can remove it.
+				if (this.isSnapshotInOldFormatAndCollectBlobs(clonedSnapshotTree, blobMap)) {
+					this.sanitizeSnapshot(clonedSnapshotTree);
+				}
+				return createChannelServiceEndpoints(
+					dataStoreContext.connected,
+					submitFn,
+					this.dirtyFn,
+					addedGCOutboundReferenceFn,
+					storageService,
+					logger,
+					clonedSnapshotTree,
+					blobMap,
 				);
-			});
-		}
-		return this.channel;
-	}
-
-	private async loadChannel(): Promise<IChannel> {
-		assert(!this.isLoaded, 0x18e /* "Channel must not already be loaded when loading" */);
-		assert(
-			await this.services.value.objectStorage.contains(".attributes"),
-			0x190 /* ".attributes blob should be present" */,
-		);
-		const attributes = await readAndParse<IChannelAttributes>(
-			this.services.value.objectStorage,
-			".attributes",
-		);
+			}),
+			new LazyPromise<IChannel>(async () => {
+				const { attributes, factory } = await loadChannelFactoryAndAttributes(
+					dataStoreContext,
+					this.services.value,
+					this.id,
+					registry,
+				);
+				const channel = await loadChannel(
+					runtime,
+					attributes,
+					factory,
+					this.services.value,
+					logger,
+					this.id,
+				).catch((err) => {
```

Comment:
to make sure I'm reading this correctly--rationale on changed error behavior is because we expect specific data processing errors from `loadChannelFactoryAndAttribues`, and we expect channel's delta handler to already wrap errors it emits in data processing errors. Right?

Reply:
any reason why we don't want this behavior for the analogous calls in `remoteChannelContext.ts`?

Reply:
i should probably wrap both loadChannelFactoryAndAttribues and loadChannel. i was just matching the existing behavior. i believe it wraps for observability reasons only specifically they want to know if this particular path has problems for whatever reason.